### DISCUSSION TITLE
SFDX Org Browser is missing metadata types (profiles, InstalledPackage, and more)

### DISCUSSION THREAD

**Discussion by peternoges** [forcedotcom/salesforcedx-vscode#5784] [Ideas]
### Summary
**Recap:**
- Some items are missing from orgbrowser tree like InstalledPackage, ProfileSessionSetting, ProfilepasswordPolicies, and Profiles
- Additionally, im removing standard obj from the claim above,  however in a different way its somewhat-missing too;  specifically for standard objects in the orgbrowser it can lead to missed file downloads depending on where you click within orgbrowser which is written about in #2824.   Seems the CustomObject retrieve button mentioned in  #2824 doesnt cascade, perhaps?)

### Steps To Reproduce:
1.  Auth a sfdx project to an org
2. use sfdx orgbrowser. Notice its missing the types: InstalledPackage, ProfileSessionSetting, ProfilepasswordPolicies, and Profiles

### Expected result

Ideally we'd see the types from step 2 within orgbrowsers tree. 

### Actual result

The types from step 2 dont appear in orgbrowsers tree

### Additional information

December 30, 2020 Edit:  
This post questions why some metadatatypes are missing from orgbrowser's  ui tree; standard objects do appear, so i've revised the repo steps and write-up.  

**VS Code Version**:
![image](https://user-images.githubusercontent.com/7834562/97650899-07f58400-1a18-11eb-8c7d-6e6770874d8b.png)

**SFDX CLI Version**:
sfdx-cli/7.74.1 win32-x64 node-v12.16.1

**OS and version**:
Windows 10

**Comment by lcampos** [comment]
Hi @peternoges , thanks for opening this question. The answer to this is that it depends on the entity/object. I can confirm we won't be adding InstalledPackages and Profiles. 

The reason for this is that installing packages is done through the use of the cli command `sfdx force:package:installed:list` or by declaring them as dependencies on `sfdx-project.json` if you're working on a second generation package (https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev2gp_config_file.htm). 

For Profiles it's a bit different, we do not support it in Org Browser because it's not possible to retrieve a correct metadata shape for a Profile without retrieving every potential piece of metadata associated with it, that is just a limitation of the APIs.

For standard objects, we do support them but they are currently being embedded under `Custom Objects` because that's how the APIs categorize them. We could take a look at this cc @smaddox-sf .

For ProfileSessionSetting & ProfilePasswordPolicies, we might've removed those from the Org Browser cause of the way that they were supported back when Org Browser was first released but we can take a look at it.

**Comment by dvolk20** [comment]
@lcampos - with the limitation of profiles not being included in org browser, what is the best alterative?

**Comment by peternoges** [comment]
Hey @lcampos! thanks very much for the discussion.

Here's an update about the multiple requests to support more types within SFDX OrgBrowser; I've also recently included sfdx/os version info in to my original post.  Disregard 1,  but what are your thoughts on 2,3,4? 

1. Disregard original post's  inquiry about  supporting "Standard Objects" in Orgbrowser because i can no longer reproduce that. 

2. Furthermore,  I'm less concerned about ProfilepasswordPolicies and ProfileSessionSetting being supported in Orgbrowser (but why not support them?)

3. Additionally, Please consider supporting "Profile" in Orgbrowser.  ChangeSets support something similar by way of "Include Profiles", and ideally OrgBrowser would as well. It could be tooltipped to explain that each orgBrowser retrieval  of Profile retrieves ‘only the meta for the directory of force-app, and/or the meta found in package.xml’.  Similarly a "profile" retrieve in orgbrowser (if supported) ,  could perhaps by-default respect forceignore and ignore .dup files

4. Lastly, should Orgbrowser support InstalledPackage retrievals?  Would there be anything to return folder/file wise for InstalledPackage (if supported in orgbrowser) ? Maybe package version, method signatures, etc (obviously nothing protected IP, ie class bodies, vfp components, etc)

**Comment by peternoges** [comment]
Hey all - I have re-written the original post with a revised perspective of the issue.

**Comment by smaddox-sf** [comment]
Thank you @peternoges .  We are planning some work on Org Browser this year and will consider this feedback as we do.  As @lcampos mentioned above, a couple of the types you're interested in aren't possible at add but a couple might be.
